src/action-ros-ci.ts

```diff
@@ -191,14 +200,50 @@ async function run() {
 		const extraCmakeArgs = core.getInput("extra-cmake-args");
 		const colconExtraArgs = core.getInput("colcon-extra-args");
 		const importToken = core.getInput("import-token");
-		const packageNames = core.getInput("package-name", { required: true }).split(RegExp("\\s")).join(" ");
+		const packageNames = core
+			.getInput("package-name", { required: true })
+			.split(RegExp("\\s"))
+			.join(" ");
 		const rosWorkspaceName = "ros_ws";
 		core.setOutput("ros-workspace-directory-name", rosWorkspaceName);
 		const rosWorkspaceDir = path.join(workspace, rosWorkspaceName);
 		const targetRos1Distro = core.getInput(targetROS1DistroInput);
 		const targetRos2Distro = core.getInput(targetROS2DistroInput);
 		const vcsRepoFileUrlListAsString = core.getInput("vcs-repo-file-url") || "";
-		const vcsRepoFileUrlList = vcsRepoFileUrlListAsString.split(RegExp("\\s"));
+		let vcsRepoFileUrlList = vcsRepoFileUrlListAsString.split(RegExp("\\s"));
+
+		// Check if PR overrides/adds supplemental repos files
+		const vcsReposOverride = dep.getReposFilesOverride(github.context.payload);
+		const vcsReposSupplemental = dep.getReposFilesSupplemental(
+			github.context.payload
+		);
+		await core.group(
+			"Repos files: override" +
+				(vcsReposOverride.length === 0 ? " - none" : ""),
+			() => {
+				for (const vcsRepos of vcsReposOverride) {
+					core.info("\t" + vcsRepos);
+				}
+				return Promise.resolve();
+			}
+		);
+		if (vcsReposOverride.length > 0) {
+			vcsRepoFileUrlList = vcsReposOverride;
+		}
+		await core.group(
+			"Repos files: supplemental" +
+				(vcsReposSupplemental.length === 0 ? " - none" : ""),
+			() => {
+				for (const vcsRepos of vcsReposSupplemental) {
+					core.info("\t" + vcsRepos);
+				}
+				return Promise.resolve();
+			}
+		);
+		if (vcsReposSupplemental.length > 0) {
```

Comment:
nitpick: this check shouldn't be necessary, `.concat` works fine with empty lists

Reply:
removed in 13b85024e9a76d0cbc293b40092a5bb1b4cfd004